calculators/_memoria_rating.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+$GLOBALS['rating_calculators']['_memoria_rating'] = [
+	'description' => '
+		by <a href="/player/125980">Memoria</a>: Rating with per-matchup rating calculation, v1
+		
+		-Only for 1v1, 2v2, and 3v3.
+  		-Games under 4 minutes don\'t count.
+		-Just like in the main rating, people start at 2000 rating,
+		 4000 rating difference = 90% win prediction, max 100 rating per match
+  		-Rating prediction is calculated for each matchup (1v1),
+		 then it\'s combined into a final prediction 
+		',
+	'default_rating' => 2000,
+	'fields' => [
+		'g.duration',
+	],
+	'main' => '_memoria_rating__main',
+	];
+
+function _memoria_rating__main(&$teams) {
+	$game_duration_seconds = floor($teams[0][0]['duration']/22.4);
```

Comment:
you are making unrelated checks below. This one is only checked on line 27. Therefore this variable should be first set on line 26, right before it's used. No need to do this if you are going to exit because of wrong players numbers.